gtda/externals/python/ripser_interface.py

```diff
@@ -158,8 +160,56 @@ def _resolve_symmetry_conflicts(coo):
         return _row, _col, _data
 
 
+def _compute_dtm_weights(dm, n_neighbors, weights_r):
+    knn = kneighbors_graph(dm, n_neighbors=n_neighbors,
+                           metric="precomputed", mode="distance",
+                           include_self=False)
```

Comment:
If `n_neighbors` is supposed to correspond to $k_0$ from the paper, i think we should set `include_self=True`

Reply:
Thanks for looking into the formula! Out of curiosity, how do you see this? In any case, from a user perspective I would be against considering a point as its own neighbor. But if you are right we should probably divide by `n_neighbors + 1` instead?

Reply:
I agree with your suggestion.
I am looking at the top of page 13 from the paper 
![image](https://user-images.githubusercontent.com/32167173/102816198-e9838780-43cd-11eb-829d-b5ad25aea5db.png)
Here, the DTM is defined on $E=R^d$, so the first nearest neighbor can itself. Let me know if you think it's the correct interpretation. Anyway, i totally agree that it's better to just vary the constant in front !

Reply:
I agree with you and have made the change!